### PR TITLE
Add adyen driver

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Add Adyen driver compatible with Adyen Terminal API 3.0

### CODE REVIEW

**Review by sbidoul — APPROVED** [review]
LGTM

File: pywebdriver/plugins/adyen_driver.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        request_data = self._prepare_request_data(payment_info, terminal_id)
+        success = False
+        status = ""
+        psp_reference = ""
+        try:
+            response = requests.post(
+                self.endpoint, data=json.dumps(request_data), headers=request_headers
+            )
+            response.raise_for_status()
+            jsonResponse = response.json()
```
[the remark this answers is not in the document]
Comment (adrienpeiffer) [reply pywebdriver/plugins/adyen_driver.py:73]: Done

File: pywebdriver/plugins/adyen_driver.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    jsonResponse["SaleToPOIRequest"]["EventNotification"],
+                )
+        except requests.exceptions.HTTPError as errh:
+            status = "Adyen driver - HTTP Error"
+            app.logger.error("%s : %s", status, errh)
+        except requests.exceptions.ConnectionError as errc:
+            status = "Adyen driver - Error Connecting"
+            app.logger.error("%s : %s", status, errc)
+        except requests.exceptions.Timeout as errt:
+            status = "Adyen driver - Timeout Error"
```
[the remark this answers is not in the document]
Comment (adrienpeiffer) [reply pywebdriver/plugins/adyen_driver.py:98]: These error messages are not displayed in the POS. The message in the POS is based solely on the success or failure of the transaction. https://github.com/odoo/odoo/blob/c6b6dd5650fd3a4b3e6942505075803eaa30a240/addons/point_of_sale/static/src/js/Screens/PaymentScreen/PaymentScreen.js#L381

**Comment by adrienpeiffer** [comment]
pre-commit check should be ok now.

**Comment by sbidoul** [comment]
@PierrickBrun can you have a look and merge if ok for you?

**Comment by PierrickBrun** [comment]
LGTM.

I think the replacement from `jsonResponse` to `json_response` is not present yet. I think I will wait for it to be added before merging

**Comment by adrienpeiffer** [comment]
@PierrickBrun Replacement done

**Comment by adrienpeiffer** [comment]
@PierrickBrun Thanks for the quick review :smile:

**Pull request merged**